### PR TITLE
Sharing buttons update

### PR DESCRIPTION
**Pull request by Themroc**
G+ and delicious are no more available.

### CODE REVIEW

File: public/js/selfoss-shares.js [hunk 1]
```diff
@@ -8,11 +8,8 @@ selfoss.shares = {
         this.enabledShares = enabledShares;
         this.initialized = true;
 
-        this.register('delicious', 'd', function(url, title) {
-            window.open('https://delicious.com/save?url=' + encodeURIComponent(url) + '&title=' + encodeURIComponent(title));
-        });
-        this.register('googleplus', 'g', function(url) {
-            window.open('https://plus.google.com/share?url=' + encodeURIComponent(url));
+        this.register('diaspora', 'd', function(url, title) {
+            window.open('https://share.diasporafoundation.org/?url='+encodeURIComponent(url)+'&title='+encodeURIComponent(title)+'&notes='+'');
```
Comment (jtojnar) [inline public/js/selfoss-shares.js]: `+` needs to be spaced from both sides.

Comment (jtojnar) [reply public/js/selfoss-shares.js]: Also why is `'&notes='+''` needed?

Comment (Themroc) [reply public/js/selfoss-shares.js]: " + " Oh, really? Works in chromium... but no prob, will fix. And remove the &notes as well.

Comment (Themroc) [reply public/js/selfoss-shares.js]: Btw. very nice piece of software! Never seen one where such a change requires only touching one file.

Comment (jtojnar) [reply public/js/selfoss-shares.js]: Yeah, it is just a style preference – we are trying to have an uniform style everywhere and have a scripts to enforce that. If you have the development environment set up, you can use `npm run lint:client -- --fix` command to convert the code to the preferred style automatically.

Comment (Themroc) [reply public/js/selfoss-shares.js]: Travis was complaining about the "+" vs. " + ". That's fixed with another commit.
Graphics is from here:
https://www.deviantart.com/creativekaizen/art/Diaspora-Icon-Set-257241499
Vector is in .ai-file (Adobe Illustrator, i guess). Inkscape can open & save as .svg. Should i upload that somewhere?

Comment (Themroc) [reply public/js/selfoss-shares.js]: file:///src/net/selfoss-2.18/diaspora_icons/diaspora_icons_cs5.svg

Comment (Themroc) [reply public/js/selfoss-shares.js]: The upload here is crap. Doesn't even like a 10-Byte-.txt-file :(

Comment (jtojnar) [reply public/js/selfoss-shares.js]: A link should be enough.

**Comment by jtojnar** [comment]
Sorry, I accidentally closed this when rebasing.

Could you please push to your branch again and reopen the PR.

File: public/js/selfoss-shares.js (hunk 1 above)
Comment (Themroc) [reply public/js/selfoss-shares.js]: Ok, there it is:
http://www.69b.org/diaspora.svg

**Comment by Themroc** [comment]
hmmm... i don't get this github-thingie... a few minutes ago (after your rebase) it said "last commit in may 2019". In _my_ fork... So, i re-git cloned it, applied my patches, github says you force-pushed my branch, but https://github.com/SSilence/selfoss/blob/master/public/js/selfoss-shares.js is still the old version...

**Comment by Themroc** [comment]
Think i remember the linux-folks having said they don't accept pull requests on github, because "they don't work" ... Best way forward is probably closing this PR, me deleting my fork and do it all over again tomorrow...

**Comment by jtojnar** [comment]
The pull request work fine (unless I accidentally force push already merged `HEAD` like current `master` instead of my rebased `HEAD`, which closes the PR and removes maintainer’s access to the branch), I just have not merged it yet. Currently testing.

**Comment by jtojnar** [comment]
It looks like the icon does not really fit in with the other monochrome icons:

image

What do you think about https://wiki.diasporafoundation.org/File:Logo.png?

**Comment by Themroc** [comment]
Ok, great!
The icon looks good enaugh for me... but yeah, little bit mushy. I'll try to remove that shadow in inkscape.

**Comment by jtojnar** [comment]
This is how it looks with the new icon:

image

**Comment by Themroc** [comment]
How's this
http://www.69b.org/diaspora-simple.svg
 one?

**Comment by jtojnar** [comment]
I think I still prefer the black asterisk, as that is what they have on the [branding page](https://wiki.diasporafoundation.org/Branding). The other sharers (except for Facebook and Wordpress) also use positive, rather than negative images.

**Comment by jtojnar** [comment]
And also, the positive is what the icon set we will most likely use provides:

https://fontawesome.com/icons/diaspora?style=brands

**Comment by Themroc** [comment]
This
dia
is what a diaspora-pod looks like on a browser's bookmark bar, so i'd rather use white-on-black... But i have to admit, i like your black-on-white better now...

**Comment by jtojnar** [comment]
I merged it now, thanks for all your work.

**Comment by Themroc** [comment]
Great, thanks!
Now for something completely different...

Some feeds lack a <channel><image> tag, but do provide a <channel><link>, where usually a favicon can be found. Would be nice if that's used. Now, unfortunately, many pages still use the traditional .ico format, which i guess would cause lots of trouble. So, lets convert it to .png. Here
https://www.phpclasses.org/package/3906-PHP-Read-and-write-images-from-ICO-files.html
is a class to do that, but it comes with a special license. Would you accept such a PR? And/or should i open a feature request?
The license reads:

>  ***************************************************************************
>  *  Original floIcon copyright (C) 2007 by Joshua Hatfield.                *
>  *  All Rights Reserved                                                    *
>  ***************************************************************************
> 
> 							 floIcon License
> 
>                       Program & Concept created by
> 
> Joshua Hatfield
> 2368 Franklin St.
> Augusta, GA 30906
> USA
> (email flobi@flobi.com)
> 
> This document contains the rules by which you can use, alter or publish
> parts of floIcon. floIcon has been created by Joshua Hatfield in his spare
> time. You are legally bound to follow the rules described in this document.
> 
> Rules:
> 
>    !! floIcon is NOT Public Domain, shareware, careware or the like !!
> 
>    You may under no circumstances make profit on *ANY* part of floIcon in
>    any possible way. You may under no circumstances charge money for
>    distributing any part of floIcon - this includes the usual $5 charge
>    for "sending the disk" or "just for the disk" etc.  By breaking these rules
>    you violate this agreement, and hence will be sued.
> 
>    You may not remove any copyright notices from any of the documents or
>    sources given to you.
> 
>    This license must *always* be included "as is" if you copy or give
>    away any part of floIcon (which is to be done as described in this
>    document).
> 
>    If you publish *any* part of floIcon, I, as creator, must appear in the
>    article, and the article must be clearly copyrighted subject to this
>    license. Before publishing you must first send me a message, by
>    snail-mail or e-mail, and inform me what, where and when you are
>    publishing (remember to include your address, name etc.)
> 
>    Any running version of floIcon must include my name in the login and/or
>    intitial connection sequence. Furthermore the if there is a credit section
>    or command in any application using floIcon, it shall always contain my
>    name, addresses, and a notice which states I have created floIcon.
> 
>    You are allowed to alter floIcon, source and documentation as long as
>    you do not violate any of the above stated rules.
> 
> Regards,
> 
> Joshua Hatfield
> 
> 
> Note:
> 
> I hope you will enjoy floIcon, and encourage you to send me any reports on
> bugs (when you find 'it'). Remember that I am using my spare time to write and 
> improve floIcon, bugs, etc. - and changes will take their time.  I have so
> far put a number of programming hours into this project.  If you make any major 
> improvements on floIcon I would be happy to hear from you.  As you will
> naturally honor the above rules, you will receive new updates and improvements 
> made to the class.

**Comment by jtojnar** [comment]
Could you please open a new issue for that, it will allow us to track it separately.

**Pull request merged**